pkg/koordlet/runtimehooks/protocol/container_context.go

```diff
@@ -71,13 +73,26 @@ func (c *ContainerRequest) FromProxy(req *runtimeapi.ContainerResourceHookReques
 	}
 }
 
-func (c *ContainerRequest) FromReconciler(podMeta *statesinformer.PodMeta, containerName string) {
+func (c *ContainerRequest) FromReconciler(podMeta *statesinformer.PodMeta, containerName string, sandbox bool) {
 	c.PodMeta.FromReconciler(podMeta.Pod.ObjectMeta)
 	c.ContainerMeta.Name = containerName
-	for _, containerStat := range podMeta.Pod.Status.ContainerStatuses {
-		if containerStat.Name == containerName {
-			c.ContainerMeta.ID = containerStat.ContainerID
-			break
+	c.ContainerMeta.Sandbox = sandbox
+	if sandbox {
+		var err error
+		if c.ContainerMeta.ID, err = koordletutil.GetPodSandboxContainerID(podMeta.Pod); err != nil {
```

Comment:
we need to cache the cgroup path of the sandbox container in the future to reduce the overhead